### ISSUE TITLE
Cannot remove Nics when VM and NIC are in different Resource groups

### ISSUE THREAD

**Issue by mahedley** [Azure/azure-cli#4136]
### Description
The az vm nic remove command doesn't support the scenario where the nic and the virtual machine are in separate azure resource groups. This is because you are only able to specify a single resource group as input to the command.

---

### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here: Azure CLI Window

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here: 2.0.12

**OS Version:** What OS and version are you using?  
Answer here: Azure web cli

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here: bash

**Comment by tjprescott** [comment]
```
    az vm nic remove: Remove network interface configurations from the virtual machine.

Arguments
    --nics              [Required]: Names or IDs of NICs.
    --resource-group -g [Required]: Name of resource group. You can configure the default group
                                    using `az configure --defaults group=<name>`.
    --vm-name           [Required]: The name of the Virtual Machine. You can configure the default
                                    using `az configure --defaults vm=<name>`.
    --primary-nic                 : Name or id of the primary NIC. If missing, the first of the NIC
                                    list will be the primary.
```

If you have a nic in a different resource group from the VM, then you should specify the NIC by ID.  If that doesn't work, there is a bug.

**Comment by wtfiwtz** [comment]
It definitely looks like this is a bug, which I came across today... 

"az vm create" will create a subnet in the wrong resource group if you don't specify an absolute path upfront and the subnet is in a different subscription. 

When you attempt to change the subnet (to put in the correct address range) you can't do any of the following:

- Remove a NIC / IP config (must always have at least one)
- Add a second NIC / IP config in a different resource group (even after stop-deallocate of the VM)
- Set the NIC subnet directly (to force it to the correct one)

So basically you are forced to destroy and recreate the VM from scratch.

**Comment by tjprescott** [comment]
If you specify a subnet name, then the subnet must be in the same subscription you are logged in to and it must be in the same resource group your VM will be created in.  If that's not the case, then the CLI will create the subnet in that location. If you want to use an existing subnet in a different subscription or resource group, you must specify the full ARM ID.

What would say is the bug here?  That the CLI creates the subnet if it isn't found is intentional. What errors do you get when attempting to change the subnet?

There are some scenarios where the only way to "fix" an erroneous VM deployment is to destroy it and recreate it (for example, if you wanted a multi-NIC VM but created it with only one inadvertently).

**Comment by wtfiwtz** [comment]
It may or may not be a bug but its certainly a massive usability issue.

I have shut down the VM, deallocated it, and yet I cannot change *any* NIC settings through the CLI, even to attach it to another subnet.

**Comment by tjprescott** [comment]
@sauryadas is this a service-side limitation?

**Comment by sauryadas** [comment]
i think this is a service side implementation . Best to loop in NRP folks

+ @jluk

**Comment by wtfiwtz** [comment]
@jluk this seems to be an existing known issue: https://feedback.azure.com/forums/216843-virtual-machines/suggestions/4026122-ability-to-move-vm-into-a-vnet-virtual-network
